atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/ChannelPool.java

```diff
@@ -84,7 +88,14 @@ private int getChannelOffset(final String messageType) {
    * @return a future to be completed with a channel from the pool
    */
   CompletableFuture<Channel> getChannel(final Address address, final String messageType) {
-    final List<CompletableFuture<Channel>> channelPool = getChannelPool(address);
+    final InetAddress inetAddress = address.address();
+    if (inetAddress == null) {
+      final CompletableFuture<Channel> failedFuture = new OrderedFuture<>();
```

Comment:
I'm not sure. This part of the code used `OrderedFuture`, so I used it for consistency.

Reply:
Is it maybe because we want to execute all the "listeners" which listen for the future completion in order on which they have been registered? :thinking: And the normal completable futruee doesn't guarantee this ? :thinking:

Reply:
Technically yes, but I never understood why this is such a big advantage :shrug: We could likely write everything without this constraint